### PR TITLE
Add effort attribute to task

### LINKED ISSUE #81: Refactor code for tasks to have effort estimate

### PR DESCRIPTION
**Pull request by joyngjr**
Add effort attribute to tasks
Refactor code so that tasks have an effort attribute
Default value of 2 but accepts user-defined values at instantiation
Supports editing of effort attribute

Closes #81

### CODE REVIEW

File: src/test/java/seedu/task/storage/JsonAdaptedTaskTest.java [hunk 1]
```diff
@@ -136,13 +154,13 @@ public void toModelType_deadlineAndEventOverlap_throwsIllegalValueException() {
     public void toModelType_nullAlertWindow_stillSucceed() throws Exception {
         JsonAdaptedTask task =
                 new JsonAdaptedTask(VALID_NAME, VALID_DESCRIPTION, VALID_TAGS,
-                        EMPTY_DATE, EMPTY_DATE, EMPTY_DATE, null);
+                        EMPTY_DATE, EMPTY_DATE, EMPTY_DATE, VALID_EFFORT, EMPTY_WINDOW);
         JsonAdaptedTask task1 =
                 new JsonAdaptedTask(VALID_NAME, VALID_DESCRIPTION, VALID_TAGS,
-                        EMPTY_DATE, EMPTY_DATE, EMPTY_DATE, EMPTY_WINDOW);
+                        EMPTY_DATE, EMPTY_DATE, EMPTY_DATE, VALID_EFFORT, EMPTY_WINDOW);
         JsonAdaptedTask task2 =
```
Comment (lywich) [inline src/test/java/seedu/task/storage/JsonAdaptedTaskTest.java:161]: I think we should maintain the "null" of task. The change proposed would make the assertequals duplicate.

File: src/test/java/seedu/task/testutil/SimpleTaskBuilder.java [hunk 2]
```diff
@@ -77,7 +87,11 @@ public SimpleTaskBuilder withAlertWindow(String alertWindow) {
     }
 
     public Task build() {
-        return new SimpleTask(name, description, tags);
+        return new SimpleTask(name, description, tags, effort);
+    }
+
+    public Task buildDefault() {
+        return new SimpleTask(name, description, tags, null);
     }
```
Comment (lywich) [inline src/test/java/seedu/task/testutil/SimpleTaskBuilder.java:95]: null value being passed into the object seems to be dangerous. How about ```new effort()```?

**Review by lywich — COMMENTED** [review]
Nice work, I have left some comments to clarify some questions. Keep up the good work.

File: src/main/java/seedu/task/model/task/Effort.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    public Effort(long effort) {
+        if (effort < 0) {
+            throw new InvalidEffortException();
+        }
+        this.effort = effort;
+    }
+
+    public Effort() {
+        this.effort = 2;
+    }
```
Comment (lywich) [inline src/main/java/seedu/task/model/task/Effort.java:27]: How about a value that users can identify better with? I think from a user standpoint they might be curious why 2 was chosen.

File: src/test/java/seedu/task/testutil/SimpleTaskBuilder.java (hunk 2 above)
Comment (joyngjr) [reply src/test/java/seedu/task/testutil/SimpleTaskBuilder.java:95]: Good job noticing that! This is actually another method I created to test if SimpleTask automatically uses a default effort value when the user does not key it in. The regular build() method used to build other SimpleTask stubs is still present!

File: src/test/java/seedu/task/storage/JsonAdaptedTaskTest.java (hunk 1 above)
Comment (joyngjr) [reply src/test/java/seedu/task/storage/JsonAdaptedTaskTest.java:161]: Thanks! Didn't notice that I changed the test case :)

**Review by lywich — APPROVED** [review]
Nice work with the changes.

**Pull request merged**